### DISCUSSION TITLE
Nested Groups strange behavior with template literals

### DISCUSSION THREAD

**Discussion by ivanpajon** [tailwindlabs/tailwindcss#12557] [Help]
**What version of Tailwind CSS are you using?**

For example: v2.0.4

**What build tool (or framework if it abstracts the build tool) are you using?**

For example: PostCSS 8.4.30, React 18.2.0, NextJS 13.5.1, Typescript 5.2.2, Tailwind 3.3.5

**What version of Node.js are you using?**

Tested with v18.18.0 and v20.9.0

**What browser are you using?**

Chrome (latest)

**What operating system are you using?**

Windows 11

**Reproduction URL**

Reproduction working in Tailwind Play and plain HTML Elements:
https://play.tailwindcss.com/sDDG7CrsNx

Reproduction of strange behavior in Stackblitz, using Tailwind in React components (NextJS framework):
https://stackblitz.com/edit/tailwind-nested-groups-error?file=app%2Fpage.tsx

**Describe your issue**

I am developing a recursive component to display menu options. If the button shown has nested children it is shown with an arrow. I want to use [Tailwind's Nested Groups](https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups) to differentiate the animations of each sublevel and make them independent.
The name I want to give to each group depends on an id, so following [Tailwind's recommendations](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) for building dynamic classes I do it by declaring a variable to store the name of the dynamic class using template strings, and then passing the variable to className.
The result is that I tried many ways to do this but I couldn't independently animate the arrows for each level.

```jsx
const groupName = `group/${item.id}`;
const groupAction = `group-open/${item.id}:-rotate-180`;

<details
  key={item.id}
  className={`${groupName} [&_summary::-webkit-details-marker]:hidden`}
>
  <span className={`${groupAction} flex shrink-0 transition duration-300`}></span>
</details>
```

**Comment by adamwathan** [comment]
Hey! The issue here is that you are using template literals, which as mentioned in the documentation you linked will _not_ work:

<img width="807" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/4323180/7c674c4b-0bde-4b1e-b112-d4dff87e2533">

Don't have the time right this second to think through the right alternative solution for you here but hopefully others from the community can jump in with some ideas!

**Reply by ivanpajon** [reply]
But if I do something like

```jsx
const groupAction = `group-open/${item.id}:-rotate-180 flex shrink-0 transition duration-300`;

<span className={`${groupAction}`}></span>
```

Wouldn't I literally be doing this?
![image](https://github.com/tailwindlabs/tailwindcss/assets/17519714/7a81763a-f2a6-448a-af1b-358e2358a759)

**Reply by adamwathan** [reply]
The problem is this part:

```
const groupAction = `group-open/${item.id}:-rotate-180...
```

Tailwind is going to generate a class literally called `group-open/${item.id}:-rotate-180`, not `group-open/17:-rotate-180` or whatever you're expecting. Tailwind doesn't run your JavaScript, it just parses your files as plain text.
